### PR TITLE
[serve] Add telemetry for lightweight config updates

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

This change adds new telemetry to get more insight into if and how users are using the lightweight config updates feature of Serve. It adds the telemetry tags:
- `SERVE_NUM_REPLICAS_UPDATED`: whether `num_replicas` was updated (without any heavyweight option updated)
- `SERVE_USER_CONFIG_UPDATED`: whether `user_config` was updated (without any heavyweight option updated)
- `SERVE_AUTOSCALING_CONFIG_UPDATED`: whether `autoscaling_config` was updated (without any heavyweight option updated)

These values are set to True as long as the user uses the feature at least once for any deployment in any application on the cluster.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/protobuf/usage.proto [hunk 1]
```diff
@@ -60,6 +60,12 @@ enum TagKey {
   SERVE_REST_API_VERSION = 13;
   // The number of serve apps running in the cluster as a string.
   SERVE_NUM_APPS = 14;
+  // Whether num_replicas changed as a lightweight config update
+  SERVE_NUM_REPLICAS_UPDATED = 15;
+  // Whether user_config changed as a lightweight config update
+  SERVE_USER_CONFIG_UPDATED = 16;
+  // Whether autoscaling_config changed as a lightweight config update
+  SERVE_AUTOSCALING_CONFIG_UPDATED = 17;
```
Comment (edoakes) [inline src/ray/protobuf/usage.proto:68]:
```suggestion
  // Whether num_replicas changed as a lightweight config update
  SERVE_NUM_REPLICAS_LIGHTWEIGHT_UPDATED = 15;
  // Whether user_config changed as a lightweight config update
  SERVE_USER_CONFIG_LIGHTWEIGHT_UPDATED = 16;
  // Whether autoscaling_config changed as a lightweight config update
  SERVE_AUTOSCALING_CONFIG_LIGHTWEIGHT_UPDATED = 17;
```

**Review by edoakes — APPROVED** [review]
Just a nit, otherwise LGTM, do we need approval from anyone else on this?

File: src/ray/protobuf/usage.proto (hunk 1 above)
Comment (zcin) [reply src/ray/protobuf/usage.proto:68]: I believe we need approval from either @pcmoritz or @thomasdesr.

**Review by sihanwang41 — APPROVED** [review]

**Review by thomasdesr — APPROVED** [review]
Telemetry Boop

**Comment by zcin** [comment]
Failures unrelated. @edoakes ready for merge

**Review by zhe-thoughts — APPROVED** [review]
Approved for cherry picking

**Pull request merged**